orttraining/orttraining/test/training_api/core/checkpoint_test.cc

```diff
@@ -160,6 +160,36 @@ TEST(CheckpointApiTest, SaveOnnxModelAsCheckpoint_ThenLoad_CPU) {
   }
 }
 
+/**
+ * Load ONNX model with parameters set to 0 from file path, Load Checkpoint weights into the Model,
+ * Then compare the new weights to 0 to make sure they were changed after loading checkpoint to model.
+ */
+TEST(CheckpointApiTest, LoadCheckpointToModel) {
+  // Phase 1: Load a Model with weights set to zero.
+  auto model_uri = MODEL_FOLDER "transform/load_checkpoint/zero_model.onnx";
+  ONNX_NAMESPACE::ModelProto p_model;
+  ORT_ENFORCE(Model::Load(model_uri, p_model).IsOK());
+
+  // Phase 2: Load the checkpoint weights into the Model.
+  // Call Load APIs
+  PathString checkpoint_path = ORT_TSTR("testdata/transform/load_checkpoint");
+  ASSERT_STATUS_OK(LoadCheckpointToModel(checkpoint_path, p_model));
+
+  // Phase 3: Make sure the Model's weights are not equal to zero after loading the new ones.
+  // Load imported initializers into the Model
+  for (auto& init : p_model.graph().initializer()) {
+    // Convert the tensor bytes to a float vector to compare.
+    size_t len = init.raw_data().size() / sizeof(float);
+    InlinedVector<float> float_values(len);
+    std::copy(init.raw_data().data(), init.raw_data().data() + init.raw_data().size(), reinterpret_cast<char*>(&float_values.front()));
```

Comment:
should it account for the case where the data is not in raw_data?
FYI, there are helper functions that might be useful, e.g.:
https://github.com/microsoft/onnxruntime/blob/d1497bdf62b16a929762cdc71012b38aa7dbcaf1/onnxruntime/core/framework/tensorprotoutils.h#L417-L451

Reply:
is that possible? that data would not be in raw_data, could you give me an example?

Reply:
yes, for tensor data type FLOAT it is possible to use the [float_data](https://github.com/onnx/onnx/blob/55d3b80f3ea1261f3f74f2e35844c204de85835a/onnx/onnx.in.proto#L538) field too. more details in the TensorProto comments there.
though since this is a test, perhaps you can just assert that the tensor has raw_data if that is what is expected in the test data.